src/plugins/discover/public/application/view_components/canvas/discover_visualization.tsx

```diff
@@ -65,7 +64,11 @@ export const DiscoverVisualization = ({ hits, bucketInterval, chartData, rows }:
   return (
     <EuiPanel className="discoverVisualization" data-test-subj="visualizationLoader">
       {expression ? (
-        <ReactExpressionRenderer expression={expression} searchContext={searchContext} />
+        <ReactExpressionRenderer
+          key={JSON.stringify(searchContext) + expression}
```

Comment:
I don't have major concerns with this in the wip branch, but I do feel we should avoid "forcing" updates with non-functional prop updates. We can look at why the ReactExpressionRenderer isn't re-rendering properly with the change in `searchContext`, but my hunch is that it's the same object reference which wouldn't get picked up if we have shallow memoization. Happy to take a deeper look later on when we prepare these changes for main.